File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
    ListNode* temp1=headA;
    ListNode* temp2=headB;
    int count1=0,count2=0;
    while(temp1!=NULL){
    count1++;
    temp1=temp1->next;
    }
    while(temp2!=NULL){
    count2++;
    temp2=temp2->next;
    }
    ListNode* list1=headA;
    ListNode* list2=headB;
    if(count1>count2){
    int diff=count1-count2;
    while(diff){
    list1=list1->next;
    diff--;
    }
    }else{
    int diff=count2-count1;
    while(diff){
    list2=list2->next;
    diff--;
    }
    }
    ListNode* a=list1;
    ListNode* b=list2;
    while(a!=NULL && b!=NULL){
    if(a==b) return a;
    a=a->next;
    b=b->next;
    }
    return NULL;
    }
};

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        len1=0
        len2=0
        temp1=headA
        temp2=headB
        while temp1:
            temp1=temp1.next
            len1=len1+1
        while temp2:
            temp2=temp2.next
            len2=len2+1
        a=headA
        b=headB
        if len1>len2:
            diff=len1-len2
            while diff:
                a=a.next
                diff=diff-1
        else:
            diff=len2-len1
            while diff:
                b=b.next
                diff=diff-1
        while a and b:
            if a==b:
                return a
            a=a.next
            b=b.next
        return a

                
        
        